lib/DxilContainer/DxilContainerAssembler.cpp

```diff
@@ -485,18 +485,18 @@ class DxilPSVWriter : public DxilPartWriter  {
   }
 
 public:
-  DxilPSVWriter(const DxilModule &mod, uint32_t PSVVersion = 0)
+  DxilPSVWriter(const DxilModule &mod, uint32_t PSVVersion = UINT_MAX)
   : m_Module(mod),
     m_PSVInitInfo(PSVVersion)
   {
     m_Module.GetValidatorVersion(m_ValMajor, m_ValMinor);
-    // Allow PSVVersion to be upgraded
-    if (m_ValMajor == 0 && m_ValMinor == 0) {
-      // Validation disabled upgrades to maximum PSVVersion
-      m_PSVInitInfo.PSVVersion = MAX_PSV_VERSION;
-    } else if (m_PSVInitInfo.PSVVersion < 1 && (m_ValMajor > 1 || (m_ValMajor == 1 && m_ValMinor >= 1))) {
+    // Constraint PSVVersion based on validator version
+    if (PSVVersion > 0 && DXIL::CompareVersions(m_ValMajor, m_ValMinor, 1, 1) < 0)
+      m_PSVInitInfo.PSVVersion = 0;
+    else if (PSVVersion > 1 && DXIL::CompareVersions(m_ValMajor, m_ValMinor, 1, 6) < 0)
       m_PSVInitInfo.PSVVersion = 1;
-    }
+    else if (PSVVersion > MAX_PSV_VERSION)
+      m_PSVInitInfo.PSVVersion = MAX_PSV_VERSION;
```

Comment:
I realize there is no longer any code that calls this with `PSVVersion == 0`. Nevertheless, previously such a call would result in setting the version to 1 and now it just leaves it as 0.

Reply:
Yes, the changes here are intended to fix the issue where the validator is unable to generate the PSV matching an earlier version in order to validate it.  This could cause a validator to fail validation when used with a compiler that didn't support the latest version for no good reason.  The logic of this code wasn't right, so these changes are meant to fix that.